### PR TITLE
fix code sample in the Form component doc (issue #9591)

### PR DESCRIPTION
**Pull request by pellea**
* [x] Update API docs for the component.

Run `npm run lint` did not work. By @benjycui  suggestion I skipped the checks.

This PR is linked to the following issue : https://github.com/ant-design/ant-design/issues/9591

### CODE REVIEW

File: components/form/demo/global-state.md [hunk 1]
```diff
@@ -58,9 +58,9 @@ class Demo extends React.Component {
     },
   };
   handleFormChange = (changedFields) => {
-    this.setState({
-      fields: { ...this.state.fields, ...changedFields },
-    });
+    this.setState(({ fields }) => ({
+        fields: { ...fields, ...changedFields }
```
Comment (afc163) [inline components/form/demo/global-state.md]: Keep two spaces as indent.

Comment (pellea) [reply components/form/demo/global-state.md]: Do I need to make another commit and push it on my fork? (sorry for the silly question, it's my first PR)

Comment (afc163) [reply components/form/demo/global-state.md]: No, you can fix and push it in this PR.

File: components/form/demo/global-state.md [hunk 2]
```diff
@@ -58,9 +58,9 @@ class Demo extends React.Component {
     },
   };
   handleFormChange = (changedFields) => {
-    this.setState({
-      fields: { ...this.state.fields, ...changedFields },
-    });
+    this.setState(({ fields }) => ({
+      fields: { ...fields, ...changedFields }
```
Comment (yesmeck) [inline components/form/demo/global-state.md]: > Missing trailing comma  comma-dangle

**Pull request merged**